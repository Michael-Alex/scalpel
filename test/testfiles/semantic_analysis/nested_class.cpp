class a_class
{
	public:
		class a_nested_class
		{
			int i;
		};

		class another_nested_class: public a_nested_class
		{
		};
};

a_class::a_nested_class anc;

