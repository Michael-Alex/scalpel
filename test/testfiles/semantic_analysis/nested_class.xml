<namespace id="n0">
 <class name="a_class" id="c0">
  <class name="a_nested_class" id="c1" access="public">
   <destructor access="public" defined="false">
   </destructor>
   <variable name="i" access="private">
    <type>
     <fundamental_type type="int"/>
    </type>
   </variable>
  </class>
  <class name="another_nested_class" id="c2" access="public">
   <base_classes>
    <base_class id="c1" access="public"/>
   </base_classes>
   <destructor access="public" defined="false">
   </destructor>
  </class>
  <destructor access="public" defined="false">
  </destructor>
 </class>
 <variable name="anc">
  <type>
   <class id="c1"/>
  </type>
 </variable>
</namespace>
