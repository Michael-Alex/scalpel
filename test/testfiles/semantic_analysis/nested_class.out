{
  "id": 0,
  "classes":
  [
    {
      "id": 0,
      "name": "a_class",
      "classes":
      [
        {
          "id": 0,
          "name": "nested_class",
          "access": "public",
          "destructor":
          {
            "access": "public"
          },
          "variables":
          [
            {
              "id": 0,
              "name": "i",
              "access": "private",
              "type":
              {
                "fundamental type": "int"
              }
            }
          ]
        },
        {
          "id": 1,
          "name": "nested_class2",
          "access": "public",
          "base classes":
          [
            {
              "member class id": 0,
              "access": "public"
            }
          ],
          "destructor":
          {
            "access": "public"
          }
        }
      ],
      "destructor":
      {
        "access": "public"
      }
    }
  ],
  "variables":
  [
    {
      "id": 0,
      "name": "anc",
      "type":
      {
        "member class": 0
      }
    }
  ]
}
